presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeWindows.java

```diff
@@ -124,5 +139,15 @@ private static WindowNode collapseWindows(PlanNode source, WindowNode.Specificat
                     canonical.getPrePartitionedInputs(),
                     canonical.getPreSortedOrderPrefix());
         }
+
+        private static boolean dependsOn(WindowNode parent, WindowNode dependent)
+        {
+            return parent.getPartitionBy().stream().anyMatch(symbol -> dependent.getCreatedSymbols().contains(symbol))
```

Comment:
I cannot use `DependencyExtractor.extractUnique(PlanNode)` because `WindowNode` produces on output all input symbols + created symbols (these which stand for window functions).
I'd need to calculate set difference between child's output and child's child output to see what symbols were introduced by the child and see if I depend on any of them.
E.g.: for a plan: WN1 (output: A,B,C) -> WN2 (A,B) -> Projection (A)
The difference of all symbols WN1 \ WN2 = C but what I actually care about is symbol B - introduced by WN2.
So if WN1 depended on B then WN1 and WN2 are not mergeable but if WN1 depends on A then WN1 and WN2 are mergeable.

I don't want to provide a non-generic `ExpressionExtractor` method for `WindowNode` which serves only my purpose.

Instead I've rewritten this to use `DependencyExtractor` but only to resolve `FunctionCall` arguments.

Please see 8548af7f4b4fb22df0749e0ca57e4a0d8872cc45.